src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormatter.cs

```diff
@@ -41,50 +41,88 @@ internal class CSharpFormatter
                 throw new ArgumentNullException(nameof(languageServer));
             }
 
-            if (projectSnapshotManagerAccessor is null)
-            {
-                throw new ArgumentNullException(nameof(projectSnapshotManagerAccessor));
-            }
-
             if (filePathNormalizer is null)
             {
                 throw new ArgumentNullException(nameof(filePathNormalizer));
             }
 
             _documentMappingService = documentMappingService;
             _server = languageServer;
-            _projectSnapshotManagerAccessor = projectSnapshotManagerAccessor;
             _filePathNormalizer = filePathNormalizer;
+
+            var type = typeof(CSharpFormattingOptions).Assembly.GetType("Microsoft.CodeAnalysis.CSharp.Indentation.CSharpIndentationService", throwOnError: true);
```

Comment:
Seems weird that we should do it this way instead of taking it in via Dependency injection. This there a backstory on why this is prefered?